pkg/cmd/extension/command.go

```diff
@@ -172,41 +174,116 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 				return nil
 			},
 		},
-		&cobra.Command{
-			Use:   "create <name>",
-			Short: "Create a new extension",
-			Args:  cmdutil.ExactArgs(1, "must specify a name for the extension"),
-			RunE: func(cmd *cobra.Command, args []string) error {
-				extName := args[0]
-				if !strings.HasPrefix(extName, "gh-") {
-					extName = "gh-" + extName
-				}
-				if err := m.Create(extName); err != nil {
-					return err
-				}
-				if !io.IsStdoutTTY() {
-					return nil
+		func() *cobra.Command {
+			promptCreate := func() (string, extensions.ExtTemplateType, error) {
+				var extName string
+				var extTmplType int
+				err := prompt.SurveyAskOne(&survey.Input{
+					Message: "Extension name:",
+				}, &extName)
+				if err != nil {
+					return extName, -1, err
 				}
-				link := "https://docs.github.com/github-cli/github-cli/creating-github-cli-extensions"
-				cs := io.ColorScheme()
-				out := heredoc.Docf(`
+				err = prompt.SurveyAskOne(&survey.Select{
+					Message: "What kind of extension?",
+					Options: []string{
+						"Script (Bash, Ruby, Python, etc)",
+						"Go",
+						"Other Precompiled (C++, Rust, etc)",
+					},
+				}, &extTmplType)
+				return extName, extensions.ExtTemplateType(extTmplType), err
+			}
+			// TODO fix flags
+			var flagGoBinary bool
+			var flagOtherBinary bool
+			cmd := &cobra.Command{
+				Use:   "create <name>",
```

Comment:
It looks like the `name` argument is optional now, so that could be indicated by square brackets